Mathlib/MeasureTheory/Measure/WithDensity.lean

```diff
@@ -585,6 +585,55 @@ lemma SigmaFinite.withDensity_ofReal [SigmaFinite μ] {f : α → ℝ} (hf : AEM
     SigmaFinite (μ.withDensity (fun x ↦ ENNReal.ofReal (f x))) := by
   exact SigmaFinite.withDensity_of_ne_top hf.ennreal_ofReal (ae_of_all _ (by simp))
 
+section SFinite
+
+/-- Auxiliary lemma for `sFinite_withDensity`. -/
+lemma sFinite_withDensity_of_sigmaFinite_of_measurable (μ : Measure α) [SigmaFinite μ]
+    {f : α → ℝ≥0∞} (hf : Measurable f) :
+    SFinite (μ.withDensity f) := by
+  let s := {x | f x = ∞}
+  have hs : MeasurableSet s := hf (measurableSet_singleton _)
+  rw [← restrict_add_restrict_compl (μ := μ.withDensity f) hs, restrict_withDensity hs,
+    restrict_withDensity hs.compl, ← withDensity_indicator hs, ← withDensity_indicator hs.compl]
+  have h1 : SFinite (μ.withDensity (s.indicator f)) := by
+    have h_eq_sum : s.indicator f = ∑' n : ℕ, s.indicator 1 := by
+      ext x
+      rw [tsum_apply]
+      swap; · rw [Pi.summable]; exact fun _ ↦ ENNReal.summable
+      simp_rw [Set.indicator_apply]
+      split_ifs with hx
+      · simp only [Set.mem_setOf_eq, s] at hx
+        simp [hx, ENNReal.tsum_const_eq_top_of_ne_zero]
+      · simp
+    rw [h_eq_sum, withDensity_tsum (fun _ ↦ measurable_one.indicator hs)]
+    have : SigmaFinite (μ.withDensity (s.indicator 1)) := by
+      refine SigmaFinite.withDensity_of_ne_top' (measurable_one.indicator hs).aemeasurable
+        (fun x ↦ ?_)
+      simp only [Set.indicator_apply, Pi.one_apply, ne_eq]
+      split_ifs with h <;> simp [h]
+    infer_instance
+  have h2 : SigmaFinite (μ.withDensity (sᶜ.indicator f)) := by
+    refine SigmaFinite.withDensity_of_ne_top' (hf.indicator hs.compl).aemeasurable (fun x ↦ ?_)
+    simp only [Set.indicator_apply, Set.mem_compl_iff, Set.mem_setOf_eq, ite_not, ne_eq, s]
+    split_ifs with h <;> simp [h]
+  infer_instance
+
+/-- If `μ` is s-finite, then `μ.withDensity f` is s-finite. -/
+lemma sFinite_withDensity (μ : Measure α) [SFinite μ] {f : α → ℝ≥0∞} (hf : AEMeasurable f μ) :
```

Comment:
I think enabling dot notation with `Measure.sFinite_withDensity` would be nice

Reply:
Following the suggestion of @sgouezel I am turning that into an auxiliary lemma for a future instance, which means that it should almost never be used, so dot notation will not be useful.